doc/book/box/indexes.rst

```diff
@@ -28,49 +27,225 @@ Indexes have certain limitations. See details on page
-system spaces :ref:`_space <box_space-space>` and :ref:`_index <box_space-index>`
-(for details, see reference on :ref:`box.space <box_space>` submodule).
+system spaces :ref:`_space <box_space-space>` and :ref:`_index <box_space-index>`.
 
 ..  admonition:: Tip
     :class: fact
 
-    The full information about creating an index is in section
+    See full information about creating indexes, such as
+    how to create an index using the ``path`` option, or
+    how to create a functional index in our reference for
     :doc:`/reference/reference_lua/box_space/create_index`.
 
+.. _index-box_index-operations:
+
+--------------------------------------------------------------------------------
+Index operations
+--------------------------------------------------------------------------------
+
+Index operations are automatic: if a data manipulation request changes a tuple,
+then it also changes the index keys defined for the tuple.
+
+#.  For further demonstrations let's create a sample space named ``tester`` and
+    put it in a variable ``my_space``:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space = box.schema.space.create('tester')
+
+#.  Format the created space by specifying field names and types:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space:format({
+                 > {name = 'id', type = 'unsigned'},
+                 > {name = 'band_name', type = 'string'},
+                 > {name = 'year', type = 'unsigned'},
+                 > {name = 'rate', type = 'unsigned', is_nullable=true}})
+
+#.  Create the **primary** index (named ``primary``):
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space:create_index('primary', {
+                 > type = 'hash',
+                 > parts = {'id'}
+                 > })
+
+    This is a primary index based on the ``id`` field of each tuple.
+
+#.  Insert some :ref:`tuples <index-box_tuple>` into the space:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space:insert{1, 'Roxette', 1986, 1}
+        tarantool> my_space:insert{2, 'Scorpions', 2015, 4}
+        tarantool> my_space:insert{3, 'Ace of Base', 1993}
+        tarantool> my_space:insert{4, 'Roxette', 2016, 3}
+
+#.  Create a **secondary index**:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> box.space.tester:create_index('secondary', {parts = {{field=3, type='unsigned'}}})
+        ---
+        - unique: true
+          parts:
+          - type: unsigned
+            is_nullable: false
+            fieldno: 3
+          id: 2
+          space_id: 512
+          type: TREE
+          name: secondary
+        ...
+
+#.  Create a **multi-part index** with three parts:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> box.space.tester:create_index('thrine', {parts = {
+                 > {field = 2, type = 'string'},
+                 > {field=3, type='unsigned'},
+                 > {field=4, type='unsigned'}
```

Comment:
It's a bit inconsistent to use spaces around "=" at the first line and don't do that at following lines.